### PR TITLE
Release v1.6.0-rc.1

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #<ISSUE NUMBER GOES HERE>
# What's Changed

Request to merge `develop` into `main` for Release `v1.6.0-rc.1`.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by gsnider2195 — APPROVED** [review]
That's a big release!

**Review by HanlinMiao — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "nautobot"
 # Primary package version gets set here. This is used for publishing, and once
 # installed, `nautobot.__version__` will have this version number.
-version = "1.5.24"
+version = "1.6.0-rc.1"
```
Comment (bryanculver) [inline pyproject.toml:5]:
```toml
version = "1.6.0"
```

`main` should contain the releases, all others should contain pre-releases

**Review by bryanculver — CHANGES_REQUESTED** [review]
Let's wait until #4204

File: changes/4184.added
Comment (glennmatthews) [inline changes/4184.added:1]: Need to rerun towncrier before merging this.

**Pull request closed without merging**